examples/server/server.cpp

```diff
@@ -1286,7 +1287,10 @@ int main(int argc, char **argv)
                             STOP_PARTIAL);
                     }
 
-                    const std::string to_send = llama.generated_text.substr(pos, stop_pos);
+                    const std::string to_send = stop_pos == std::string::npos
+                        ? llama.generated_text.substr(pos, std::string::npos)
+                        : ""; // just don't send anything if we're not done
+
```

Comment:
Before this fix, the to_send is a whitespace when it gets stop_pos 1 from ` L`, then the sent_token_probs_index will be incorrect.